### PR TITLE
server: don't update feeshare and rewardcut on-chain if values are unchanged

### LINKED ISSUE #1167: transcoder() is called during "Set orchestrator config" even if reward cut and fee share are unchanged
**Describe the bug**

When calling `Set orchestrator config` using the `livepeer_cli`, the process will call `transcoder()` function even when the `rewardCut` and `feeShare` parameters are not changed.

This is a waste of a transaction.

**To Reproduce**

1. Run livepeer in `-orchestrator` and `-transcoder mode`
2. Run livepeer_cli and `Invoke multi-step "become an orchestrator"`
3. See that it calls `transcoder` (among other functions)
4. Then run `Set orchestrator config`
5. Use the same parameters for `rewardCut` and `feeShare` as in step 2
6. See that it calls `transcoder` again.

**Expected behavior**

I expect that if the `rewardCut` and `feeShare` parameters entered into `Set orchestrator config` are the same as those already configured onchain, then the process shouldn't call `transcoder` again.

**Desktop (please complete the following information):**

```
livepeer-cli version 0.5.0
```
```
Livepeer Node Version: 0.5.0
Golang runtime version: gc go1.13
Architecture: amd64
Operating system: linux
```

### PR DESCRIPTION
**Pull request by kyriediculous**
**What does this pull request do? Explain your changes. (required)**

This PR prevents the webserver from submitting a `BondingManager.transcoder(rewardcut, feeshare)` transaction to set these values on-chain when the values itself aren't changed. This prevents unnecessary gas expenditure. 

**Specific updates (required)**
- Moved the RPC call to `BondingManager.getTranscoder` above the call to `BondingManager.transcoder`
- Added a check whether feeshare and rewardcut are the same as the on-chain values, only send a tx when either value or both values are changed

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**
Fixes #1167 

**Checklist:**
- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: server/webserver.go [hunk 1]
```diff
@@ -363,32 +363,34 @@ func (s *LivepeerServer) cliWebServerHandlers(bindAddr string) *http.ServeMux {
...
 		if err != nil {
 			glog.Error(err)
 			return
 		}
 
-		err = s.LivepeerNode.Eth.CheckTx(tx)
-		if err != nil {
-			glog.Error(err)
-			return
+		if t.RewardCut.Cmp(eth.FromPerc(blockRewardCut)) != 0 && t.FeeShare.Cmp(eth.FromPerc(feeShare)) != 0 {
```
Comment (yondonfu) [inline server/webserver.go]: We still want to submit the tx if one of commission rates is not equal to its on-chain value.

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: server/webserver.go (hunk 1 above)
Comment (kyriediculous) [reply server/webserver.go]: fixed in 98479bae

**Review by yondonfu — COMMENTED** [review]
Changes look good. Let's rebase

**Comment by kyriediculous** [comment]
rebased !

File: server/webserver.go [hunk 2]
```diff
@@ -363,32 +363,34 @@ func (s *LivepeerServer) cliWebServerHandlers(bindAddr string) *http.ServeMux {
...
+			}
+
+			err = s.LivepeerNode.Eth.CheckTx(tx)
+			if err != nil {
+				glog.Error(err)
+				return
+			}
 		}
 
+		glog.Infof("Setting orchestrator commission rates %v", s.LivepeerNode.Eth.Account().Address.Hex())
```
Comment (yondonfu) [inline server/webserver.go]: Could we actually move this log statement into the above conditional before the tx is submitted? As-is, this log statement would imply that the commission rates are in the process of being set *after* the tx is already submitted and confirmed on-chain.

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: server/webserver.go (hunk 2 above)
Comment (kyriediculous) [reply server/webserver.go]: fixed in bb83638d

**Review by yondonfu — APPROVED** [review]
LGTM - good to merge after a rebase

**Pull request merged**